addons/udes_priorities/models/priority.py

```diff
@@ -84,45 +105,86 @@ def _has_duplicate_names_for_same_pick_type(self):
     @api.one
     def _check_for_duplicate_names_for_same_pick_type(self):
         if self._has_duplicate_names_for_same_pick_type() > 0:
-            raise UserError("Name must be unique for the picking types it is shown on")
+            raise UserError("Name must be unique for the transfer types it is shown on")
 
-    @api.multi
-    def _number_of_oustanding_pickings(self):
-        Picking = self.env["stock.picking"]
-        self.ensure_one()
-        return Picking.search_count(
-            [("priority", "=", self.reference), ("state", "not in", ("cancel", "done")),]
+    def _is_trusted_user(self):
+        return self.env.uid == SUPERUSER_ID or self.env.user.has_group(
+            "udes_security.group_trusted_user"
         )
 
-    @api.multi
-    def _check_for_oustanding_pickings(self):
+    def _get_outstanding_pickings(self, reference_id, pick_type_id=None):
+        Picking = self.env["stock.picking"]
+        if pick_type_id:
+            return Picking.search_count(
+                [
+                    ("priority", "=", reference_id),
+                    ("picking_type_id", "=", pick_type_id),
+                    ("state", "not in", ("cancel", "done")),
+                ]
+            )
+        else:
+            return Picking.search_count(
+                [
+                    ("priority", "=", reference_id),
+                    ("state", "not in", ("cancel", "done")),
+                ]
+            )
 
-        install_mode = self.env.context.get("install_mode", False)
+    def _is_install_mode(self):
+        return self.env.context.get("install_mode", False)
 
-        if install_mode:
+    @api.multi
+    def _validate_picking_types_changes(self, values):
+        if self._is_install_mode():
             return
 
-        priorities_with_outstanding_picks = self.browse()
-        for priority in self:
-            if priority._number_of_oustanding_pickings() > 0:
-                priorities_with_outstanding_picks |= priority
-
-        if priorities_with_outstanding_picks:
-            raise UserError(
-                _("There are outstanding pickings for priorities: {}").format(
-                    ", ".join(priorities_with_outstanding_picks.mapped("name"))
-                )
+        if "picking_type_ids" in values:
+            PickingType = self.env["stock.picking.type"]
+            message = (
+                "Invalid changes, you cannot remove the following transfer types as there are outstanding "
+                "transfers\n "
             )
+            invalid = False
+            old_ids = self.picking_type_ids.ids
+            new_ids = values["picking_type_ids"][0][2]
+
+            for old_pick_type_id in old_ids:
+                if old_pick_type_id not in new_ids:
+                    outstanding_pickings = self._get_outstanding_pickings(
```

Comment:
We shouldn't be able to remove a priority if it has been set on any previous picking, not just outstanding pickings.

Reply:
After some discussions, agreed and done in latest commit.